QUESTIONS.md

```diff
@@ -0,0 +1,297 @@
+    const result = []
+    
+    for (let i = 0; i < radius.length; i++) {
+        result.push(expression(radius[i]))
+    }
+    
+    return result;
+}
+
+console.log(calculate(radius, area))
+```
+**What is the difference between filter and find?**
+
+With filter we can return an array that contain the element that satisfies the condition, but find returns an element that satisfies the condition
+
+The main different is what it return each method
+
+*Example using 'filter':*
+
+```
+const array = [1, 2, 3, 4, 5]
+
+const pair = array.filter(element => element % 2 == 0)
+
+console.log(pair)
+```
+
+*Example using 'find':*
+
+```
+const people = [
+    {
+        id: 1,
+        name: "Pablito"
+    },
+    {
+        id: 2,
+        name: "Juanito"
+    },
+    {
+        id: 3,
+        name: "Fulanito"
+    }
+]
+
+const person = people.find(element => element.id === 2)
+
+console.log(person)
+```
+
+**What does it mean to pass variables by value?**
+
+It means that the data associated with the variable is actually copied when is passed to a function, and any modification that we made in that function is gonna exists only in that function.
+
+```
+function squareArea(side) {
+    side = Math.pow(2, side)
+    
+    return side;
+}
+
+let number = 2;
+let result = squareArea(number);
+
+console.log(result);
+console.log(number);
+```
+
+**What does it mean to pass variables by reference?**
+
+Pass by reference means that the value that you're passing in a function is gonna be affected when is returned, because your'e sending its location in memory so, is not a copy, is the value that is being sending
+
+```
+function passingByReference(variable) {
+    return variable.x = 10;
+}
+
+let variable = { x: 50 } 
+
+console.log(passingByReference(variable))
+console.log(variable)
+```
+**How can you copy objets and nested objects?**
+
+In JS there is difference posibilities to copy an object, here is an example:
+
+```
+const user = {
+    name: 'Pablito', 
+    location: 'El Salvador',
+    age: 34
+}
+
+// You can use spread
+{ ...user }
+
+// Object.assing
+Object.assign({}, user)
+
+// JSON
+JSON.parse(JSON.stringify(user))
+
+```
+
+**What does immutability mean?**
+
+It means that you cannot change the value wihtout creating an entirely new value
+
+```
+let user = {
+    name: "Pablito",
+    location: "El Salvador",
+    age: 34
+};
+
+let newUser = user;
+
+const updateAge = (data, newAge) => {
+    return {
+        ...data, 
+        age: newAge
+    }
+}
+
+console.log(updateAge(newUser, 35))
+```
+**What is OOP?**
+
+It means object oriented programming and is a style based on programming with objects and classes. It has properties and actions. 
+
+It has 4 principles: 
+
+* Abstraction
+* Encapsulation
+* Inheritance
+* Polymorphism
+
+```
+class User {
+    name = "Pablito",
+    age = 34, 
+    location = "El Salvador"
+}
+
+// Actions
+
+study() {
+    ....
+}
+
+work() {
+    ....
+}
+
+eat () {
+    ....
+}
+```
```

Comment:
you need to explain more of the principles, at least some detailed examples would be great